src/index/namehash.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2019 Daniel Kraft
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_INDEX_NAMEHASH_H
+#define BITCOIN_INDEX_NAMEHASH_H
+
+#include <index/base.h>
+#include <script/script.h>
+#include <uint256.h>
+
+#include <memory>
+
+/** Default value for the -namehashindex argument.  */
+static constexpr bool DEFAULT_NAMEHASHINDEX = false;
+
+/** Maximum size of the DB cache for the name-hash index.  */
+static constexpr int64_t MAX_NAMEHASH_CACHE = 1024;
+
+/**
+ * This keeps an index of SHA-256-hashes of names to the corresponding preimage
```

Comment:
If I'm reading the code correctly, `SHA-256` should be `SHA-256d` in this comment.

Reply:
Yes indeed, thanks for pointing this out!